### DISCUSSION TITLE
share data between apps

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#4155] [Ideas]
Hi budibase,

is it possible to share data between apps using the budibase DB?

Best T

**Comment by mjashanks** [comment]
Hey @tobehn
It's not possible right now, but this might be achieved in the following ways soon:

1. We will be officially supporting the Budibase API (next few months). You will be able to connect to other apps, using a REST connector.
2. Later this year, we will add Global datasources, i.e. datasources that can be shared between apps. This will most likely be an enterprise (paid) feature.

**Reply by tobehn** [reply]
Hey @mjashanks 
Thanks for your answer. This sounds great. 

Can i buy enterprise feature for a selfhosted setup later on? Sorry if i have asked this before. I definitely would do and need that option.

**A bit offtopic:**
Just this morning i thought about a field value i would need in several apps or data providers within an app. Besides a global datasource a global data provider within an app would be awesome or would the global datasource also be present in each configured data provider as i woulnd't want to reconfigure the data provider for a global value each and every time. 

:rocket:

**Reply by mjashanks** [reply]
Of course you can pay for it when it comes!

I think I know what you are getting at, and it's something that we have discussed. We need 2 features...
1. Reusable components. I.e. the ability for the app developer to create a component structure and "Save" it, so it can be reused in multiple places
2. Global Reusable components. The ability to do Step1, that uses global datasources and can be used across many apps.

Is that what you are asking for?

**Reply by tobehn** [reply]
Well, indeed reusable components or even global reusable components could be really great.

The thing i  am looking for is a kind of global variabel context in an app. 
Lets say i want to set an app wide currency sign for string concatination i.e. €. And in case i want to stringify a value that is meant to be a price in Euro i can use the globally set currency sign by just referencing _$("appDataProvider.currencySign")_ instead of creating a Data Provider each time i want to concat the currency sign in an JS transformation. And also avoid hard coding the currency character.

If i think further ther could also be a _$("globalDataProvider")_ in order to set "system wide" values like a never changing company name.
I then could use the variables within all apps in the designer.

Does this make sense?

**Reply by mjashanks** [reply]
Ah gotcha - makes complete sense to me, I think that would be super useful.

**Reply by tobehn** [reply]
lets say in an app it would be a configurable datasource  (_add datasource -> app context_) which simply holds key:value pairs and could be edited in an interface where you can:

- add new key:value pair
- edit key
- edit value

system wide this same interface could be used in the _builder > portal > apps > global context_

The key is: As soon as you have it configured its either accessible in the specific app from anywhere or within all your apps.
